leaderboard/leaderboard.go

```diff
@@ -511,6 +511,46 @@ func (lb *Leaderboard) GetAroundMe(memberID string, order string, getLastIfNotFo
 	return members, nil
 }
 
+// GetAroundRank returns a page of results centered in the member with the given ID
+func (lb *Leaderboard) GetAroundRank(rank int, order string) ([]*Member, error) {
+	l := lb.Logger.With(
+		zap.String("operation", "GetAroundRank"),
+		zap.String("leaguePublicID", lb.PublicID),
+		zap.String("rank", strconv.Itoa(rank)),
+	)
+
+	if order != "desc" && order != "asc" {
+		order = "desc"
+	}
+
+	totalMembers, err := lb.TotalMembers()
+	if err != nil {
+		return nil, err
+	}
+
+	startOffset := rank - (lb.PageSize / 2)
```

Comment:
If I got it right, you are receiving a score in rank argument from GetAroundRankHandler and then subtracting and adding offsets to it, this logic seems wrong...
If you want players around a certain score you should first find the position of a player that has this score and then use this kind of logic.

Reply:
@Dechichi01 actually it would be simple if you did something like this:

Get a member with the given score:
```  
ZREVRANGEBYSCORE leaderboardId score -inf LIMIT 1
```

and use the returned memberId in the `GetAroundMe` method